### PR TITLE
JMAP Email/send create should use tooLarge when attempt to create an oversize mail

### PR DESCRIPTION
**Pull request by vttranlina**
Relate https://github.com/linagora/james-project/issues/4492

### CODE REVIEW

**Comment by vttranlina** [comment]
Require: update james-project when https://github.com/apache/james-project/pull/918 merged

**Review by Arsnael — APPROVED** [review]

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraEmailSendMethodContract.scala [hunk 1]
```diff
@@ -1,9 +1,5 @@
 package com.linagora.tmail.james.common
 
-import java.nio.charset.StandardCharsets
-import java.time.Duration
-import java.util.concurrent.TimeUnit
-
```
[the remark this answers is not in the document]
Comment (quantranhong1999) [reply tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraEmailSendMethodContract.scala:6]: That should reduce headache rebased lines in the future.

**Review by quantranhong1999 — APPROVED** [review]

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/EmailSendMethod.scala [hunk 2]
```diff
@@ -203,7 +206,10 @@ class EmailSendMethod @Inject()(emailSetSerializer: EmailSetSerializer,
         mailImpl.setMessageNoCopy(message)
         mailImpl
       })
-      _ <- SMono(queue.enqueueReactive(mail)).`then`(SMono.just(submissionId))
+      _ <- SMono(queue.enqueueReactive(mail))
+        .`then`(SMono.fromCallable(() => LifecycleUtil.dispose(mail))
+          .subscribeOn(Schedulers.elastic()))
+        .`then`(SMono.just(submissionId))
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/EmailSendMethod.scala:212]: While we are at it, LifecycleUtil::dispose is never called on this email. Can we fix that?

**Review by chibenwa — COMMENTED** [review]
Good initiative!

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/EmailSendMethod.scala (hunk 2 above)
Comment (vttranlina) [reply tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/EmailSendMethod.scala:212]: I don't understand what you mean. 
I just checked, LifecycleUtil.dispose is still invoked when enqueueing

**Review by chibenwa — APPROVED** [review]

**Comment by Arsnael** [comment]
Rebase please :)

**Comment by chibenwa** [comment]
Still fixup!

**Pull request merged**